tests/fgpyo/sam/test_cigar.py

```diff
@@ -31,3 +31,68 @@ def test_bad_index_raises_index_error(index: int) -> None:
 def test_bad_index_raises_type_error(index: int) -> None:
     with pytest.raises(TypeError):
         cigar[index]
+
+
+@pytest.mark.parametrize(
+    ("cigar_string", "expected_range"),
+    [
+        ("10M", (0, 10)),
+        ("10M10I", (0, 20)),
+        ("10X10I", (0, 20)),
+        ("10X10D", (0, 10)),
+        ("10=10D", (0, 10)),
+        ("10S10M", (10, 20)),
+        ("10H10M", (0, 10)),
+        ("10H10S10M", (10, 20)),
+        ("10H10S10M5S", (10, 20)),
+        ("10H10S10M5S10H", (10, 20)),
+        ("76I", (0, 76)),
+    ],
+)
+def test_query_alignment_offsets(cigar_string: str, expected_range: tuple) -> None:
```

Comment:
Please fix the type hints here, as well. Please try to avoid generic collection type hints without a parametrized type.

```suggestion
def test_query_alignment_offsets(cigar_string: str, expected_range: Tuple[int, int]) -> None:
```